docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Create token credentials from configuration
+description: This article describes how to create token credentials from configuration files
+ms.topic: how-to
+ms.custom: devx-track-dotnet, engagement-fy23
+ms.date: 1/16/2024
+---
+
+# Create Azure credential types using configuration files
+
+The `Microsoft.Extensions.Azure` library supports creating different <xref:Azure.Core.TokenCredential?displayProperty=fullName> types from values defined in _appsettings.json_ and other configuration files. This article describes the support for different `TokenCredential` types and how to configure the required values for each type.
+
+## Support for Azure credentials through configuration
+
+The `Microsoft.Extensions.Azure` library can automatically provide Azure service clients with a `TokenCredential` class by searching _appsettings.json_ or other configuration files for credential values using the default `IConfiguration` service for .NET. This approach allows developers to explicitly set credential values across different environments through configuration rather than through app code directly.
+
+The following credential types are supported via configuration:
+
+* <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>
+* <xref:Azure.Identity.WorkloadIdentityCredential?displayProperty=fullName>
+* <xref:Azure.Identity.ClientSecretCredential?displayProperty=fullName>
+* <xref:Azure.Identity.ClientCertificateCredential?displayProperty=fullName>
+* <xref:Azure.Identity.DefaultAzureCredential?displayProperty=fullName>
+
+The configuration file values are only used when the service client doesn't explicitly set an authentication mechanism. For example, the following code uses `IConfiguration` to search for values in _appsettings.json_ at run time because the <xref:Azure.Storage.Blobs.BlobServiceClient?displayProperty=fullName> does not specify credentials during instantiation:
+
+```csharp
+// No TokenCredential or access key provided - configuration files will be searched
+var blobServiceClient = new BlobServiceClient("<storage-account-name>");
+```
+
+In comparison, the following code provides `DefaultAzureCredential` directly and therefore does *not* initiate a search for configuration values in _appsettings.json_:
+
+```csharp
+// Configuration files will not be searched - DefaultAzureCredential is already provided
+var blobServiceClient = new BlobServiceClient("<storage-account-name>", new DefaultAzureCredential());
+```
+
+> [!NOTE]
+> The examples in this article use `BlobServiceClient`, but the concepts apply to other Azure service clients as well, such as `CosmosClient` or `SecretClient`.
+
+### Create a `ManagedIdentityCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>:
+
+```json
+{
+  "credential": "managedidentity",
+  "resourceId":  "<managedIdentityResourceId>"
```

Comment:
I envision two subheadings in this section:
1. User-assigned managed identity
1. System-assigned managed identity

We'll need to test this, but I think those JSON snippets would look as follows.

For user-assigned support, choose one of the following approaches:

1. Authenticate a client ID:

```json
{
  "credential": "managedidentity",
  "clientId": "<clientId>"
}
```

2. Authenticate a resource ID (the resource ID takes the form `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}`):

```json
{
  "credential": "managedidentity",
  "resourceId": "<managedIdentityResourceId>"
}
```

For system-assigned support:

```json
{
  "credential": "managedidentitycredential"
}
```

See https://github.com/Azure/azure-sdk-for-net/blob/08ac84b398ca9d6f87a70c0d5c8ec8ea9acec191/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L108-L121 for context.